sycl/test/warnings/warnings.cpp

```diff
@@ -4,6 +4,14 @@
 #include <sycl/sycl.hpp>
 
 using namespace sycl;
+
+
+class FunctorModern {
+public:
+  [[sycl::reqd_sub_group_size(16)]] void operator()() const {}
```

Comment:
My understanding is that this test main aim is to check SYCL headers. Attributes are part of the compiler and therefore I don't think that this addition worth it here. Essentially, the fact that the attribute does not produces a warning, should be checked by `clang/` tests